packages/btcindexer/src/btcindexer.ts

```diff
@@ -439,8 +446,14 @@ export class Indexer implements Storage {
 				const updates = processedPrimaryKeys.map((p) =>
 					setMintedStmt.bind(suiTxDigest, now, p.tx_id, p.vout),
 				);
-				// TODO: add logic for handling the results and console.error if failure
-				await this.d1.batch(updates);
+				try {
+					await this.d1.batch(updates);
```

Comment:
are you sure this is throwing exception rather than having errors in the return?